### ISSUE TITLE
Accessibility: Inputs should have "name"

### ISSUE THREAD

**Issue by lyubomir-popov** [canonical-web-and-design/vanilla-framework#3216]
Requirement:

![image](https://user-images.githubusercontent.com/2741678/91179789-e0640180-e6de-11ea-8f41-75c7413ce66a.png)
![image](https://user-images.githubusercontent.com/2741678/91179714-c9bdaa80-e6de-11ea-9bf4-54f5de0f368d.png)
(I do not understand the second note. MDN seems to use exactly the name attribute). 

Our example:

![image](https://user-images.githubusercontent.com/2741678/91179740-d17d4f00-e6de-11ea-9fca-53c6061278d5.png)

For comparison, MDN example:

![image](https://user-images.githubusercontent.com/2741678/91179670-bc082500-e6de-11ea-8920-e01857d52fc2.png)

**Comment by lyubomir-popov** [comment]
- Verify what the note means - about not relating to the name attr.

**Comment by lyubomir-popov** [comment]
According to [this answer](https://stackoverflow.com/questions/38477954/if-an-html-element-has-role-button-should-it-also-have-the-attribute-name/#answer-38481068), name is just a way to refer to whatever serves as aria-label, or the value of the element that has `for="{current input id}"`.

**Comment by lyubomir-popov** [comment]
The name attribute is not mandatory: [spec](https://html.spec.whatwg.org/multipage/form-control-infrastructure.html#naming-form-controls:-the-name-attribute)

Distinction between IDL attributes and content attributes:

https://stackoverflow.com/questions/12354918/what-does-idl-attribute-mean-in-the-whatwg-html5-standard-document#:~:text=Some%20IDL%20attributes%20are%20defined,attribute%20to%20the%20given%20value.

**Comment by lyubomir-popov** [comment]
Work done in https://github.com/canonical-web-and-design/vanilla-framework/pull/3307, closing
